src/gaxios.ts

```diff
@@ -154,7 +154,10 @@ export class Gaxios {
 
     opts.url = parsedUrl.href;
 
-    if (typeof options.maxContentLength === 'number') {
+    if (
+      (!opts.data || !this.isReadableStream(opts.data)) &&
```

Comment:
So I'm a little confused :) This is really only going to be passed onto `node-fetch` if the user sets the value.  This library just acts as a passthrough.  As a user, I'd be really surprised if I tried setting `maxContentLength`, only to have it not actually set because I also posted data.

Reply:
Yeah I hear you. This is avoids a bug introduced in `node-fetch@2.5.0` related to improperly using the [`size` property](https://github.com/bitinn/node-fetch/pull/637) with readable streams. My thinking was that either `gaxios` should only set the size parameter when it's useful for the request body or `node-fetch` should not use `size` to set `Content-Length` as it does in `2.5.0`.

Reply:
Ah, that makes sense.  While  I *really* appreciate the PR for this one - I think this is something that should be fixed upstream in `node-fetch`.  I'm not really comfortable making a change like this here.